djgentelella/widgets/files.py

```diff
@@ -22,17 +29,39 @@ def __init__(self, attrs=None, extraskwargs=True):
 
     def format_value(self, value):
         """File input never renders a value."""
-        return value
+        if value:
+            name = Path(value.name).name
+            self.value = mark_safe(
+                '{"name": "%(name)s", "display_name": "%(display_name)s", "url": "%(url)s" }' % {
+                    'name': value.name,
+                    'display_name': name,
+                    'url': value.url})
+            return self.value
+        return ''
+
+    def parse_value(self, value):
+        dev = None
+        try:
+            dev = json.loads(value)
+            if not ('url' in dev or 'token' in dev or 'actions' in dev):
+                dev = None
+        except json.JSONDecodeError as e:
+            logger.warning("Json error parsing: " + repr(value))
```

Comment:
```suggestion
        except json.JSONDecodeError:
            logger.exception("Json error parsing: " + repr(value))
```
Instead of using warning it is highly recommended to use `exception` as it automatically will report the trace associated with the error and even the line of code where the exception happened.